### ISSUE TITLE
False positive File is not `gofmt`-ed with `-s` (gofmt)

### ISSUE THREAD

**Issue by lucastheisen** [golangci/golangci-lint#2418]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

Getting

> File is not `gofmt`-ed with `-s` (gofmt)

When `diff -u <(gofmt -s pkg/runner/runner_unix_test.go) pkg/runner/runner_unix_test.go` reports no differences.  Also tried `gofmt -s -w pkg/runner/runner_unix_test.go` to fix, but that did not change the outcome

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.43.0 built from 861262b7 on 2021-11-03T11:57:46Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
cat: .golangci.yml: No such file or directory
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.15.3 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/vscode/.cache/go-build"
GOENV="/home/vscode/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/go/pkg/mod"
GONOPROXY="*.mitre.org"
GONOSUMDB="*.mitre.org"
GOOS="linux"
GOPATH="/go"
GOPRIVATE="*.mitre.org"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build348627259=/tmp/go-build -gno-record-gcc-switches"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /opt/app /opt / /home/vscode]
INFO [config_reader] Used config file ../../.golangci.yml
INFO [lintersdb] Active 30 linters: [asciicheck bidichk deadcode durationcheck errcheck errorlint exhaustive exportloopref gochecknoinits goconst gocritic gofmt gosec gosimple govet ineffassign nakedret nilerr nolintlint revive sqlclosecheck staticcheck structcheck typecheck unparam unused varcheck wastedassign whitespace wrapcheck]
INFO [loader] Go packages loading at mode 575 (compiled_files|deps|types_sizes|exports_file|files|imports|name) took 199.1273ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 823.4µs
INFO [linters context/goanalysis] analyzers took 5.7473914s with top 10 stages: buildir: 3.5723956s, exhaustive: 275.6197ms, inspect: 204.2263ms, ctrlflow: 169.1692ms, printf: 161.3416ms, bidichk: 155.5848ms, buildssa: 149.8764ms, the_only_name: 143.7374ms, fact_deprecated: 143.2202ms, wastedassign: 134.4311ms
INFO [runner] Issues before processing: 10, after processing: 1
INFO [runner] Processors filtering stat (out/in): nolint: 1/3, max_from_linter: 1/1, source_code: 1/1, path_prefixer: 1/1, skip_files: 10/10, path_shortener: 1/1, exclude-rules: 3/10, path_prettifier: 10/10, skip_dirs: 10/10, identifier_marker: 10/10, exclude: 10/10, uniq_by_line: 1/1, cgo: 10/10, autogenerated_exclude: 10/10, diff: 1/1, max_per_file_from_linter: 1/1, max_same_issues: 1/1, severity-rules: 1/1, sort_results: 1/1, filename_unadjuster: 10/10
INFO [runner] processing took 1.0034ms with stages: nolint: 359.5µs, exclude-rules: 236.5µs, identifier_marker: 228.4µs, autogenerated_exclude: 76µs, path_prettifier: 49.4µs, source_code: 22.5µs, skip_dirs: 19.7µs, uniq_by_line: 2.2µs, max_same_issues: 2.1µs, cgo: 1.6µs, max_per_file_from_linter: 1µs, filename_unadjuster: 1µs, path_shortener: 1µs, max_from_linter: 1µs, diff: 300ns, severity-rules: 300ns, skip_files: 300ns, exclude: 300ns, sort_results: 200ns, path_prefixer: 100ns
INFO [runner] linters took 2.3711126s with stages: goanalysis_metalinter: 2.3699089s
pkg/runner/runner_unix_test.go:1: File is not `gofmt`-ed with `-s` (gofmt)
// +build linux
INFO File cache stats: 1 entries of total size 553B
INFO Memory: 27 samples, avg is 191.7MB, max is 323.1MB
INFO Execution took 2.5781452s
```

</details>

### Code example or link to a public repository

<details>

```go
// +build linux

package runner_test

import (
	"strings"
	"testing"

	"gitlab.mitre.org/org-mitre-caasd-tools/runner/pkg/runner"
)

func Test_UnixRunForStdoutWithStdin(t *testing.T) {
	expected := "hello, world"
	result, err := runner.Run(
		runner.Config{
			PipeStdin:    strings.NewReader(expected),
			BufferStdout: true,
		},
		runner.NewCommand("cat"))
	if err != nil {
		t.Errorf("echo failed")
	}
	actual := result.Stdout.String()
	if expected != strings.TrimRight(actual, "\r\n") {
		t.Errorf("expected [%s], got [%s]", expected, actual)
	}
}

```

</details>

**Comment by ldez** [comment]
Hello,

gofmt reports an issue with the build directives in your code, the following code illustrates the fix:

```go
//go:build linux
// +build linux
```

you can call `golangci-lint run --fix` to automatically fix these kind of issue.
